### PR TITLE
Add linter and formatter

### PR DESCRIPTION
**Pull request by RadenAR**
You may get errors after merging this to master and pulling it to your local machine. Just run the following:
- `yarn lint` to run the linter and see the errors
- `yarn format` formats using prettier all the files in the repo

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/node_modules
```
Comment (RadenAR) [inline .eslintignore:1]: This is ignoring so we don't lint node modules

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "no-prototype-builtins": "off",
+    "no-unused-vars": ["error", { argsIgnorePattern: "^_" }],
+  },
+  globals: {
+    console: "readonly",
+    document: "readonly",
+    module: "readonly",
+    window: "readonly",
+  },
+}
```
Comment (RadenAR) [inline .eslintrc.js:40]: This file is settings for ESLint which is the linter I added

File: package.json [hunk 3]
```diff
@@ -16,7 +22,9 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "format": "prettier --write \"*.js\" \"**/*.js\"",
+    "lint": "eslint . --color"
   },
```
Comment (RadenAR) [inline package.json:28]: Adding scripts here so we just have to run `yarn lint` and `yarn format`

File: package.json [hunk 4]
```diff
@@ -6,6 +6,12 @@
     "@testing-library/jest-dom": "^5.11.4",
     "@testing-library/react": "^11.1.0",
     "@testing-library/user-event": "^12.1.10",
+    "eslint-config-prettier": "^8.1.0",
+    "eslint-plugin-babel": "^5.3.1",
+    "eslint-plugin-jest-dom": "^3.6.5",
+    "eslint-plugin-prettier": "^3.3.1",
+    "eslint-plugin-react": "^7.22.0",
+    "prettier": "^2.2.1",
```
Comment (RadenAR) [inline package.json:14]: Most of these are just plugins for different things we use. I didn't know these off the top of my head. The main two are eslint and prettier

File: prettier.config.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// https://prettier.io/docs/en/options.html
+module.exports = {
+  trailingComma: "es5",
+  bracketSpacing: true,
+  tabWidth: 2,
+  semi: false,
+  singleQuote: false,
+  arrowParens: "always",
+}
```
Comment (RadenAR) [inline prettier.config.js:9]: This is the settings for things like semicolons, single-quotes etc for prettier which is the formatter

File: src/App.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
-import Game from './components/game'
+import React from "react"
```
Comment (RadenAR) [inline src/App.js:1]: This although not necessary to run the code is needed for the linter 😅

**Review by thisisharrison — APPROVED** [review]
Awesome!😃

**Pull request merged**